web-common/src/features/dashboards/url-state/url-state.spec.ts

```diff
@@ -0,0 +1,511 @@
+          explore,
+          basePreset,
+        );
+
+        expect(url.toString()).to.eq(expectedUrl);
+
+        // get back the entity from url params
+        const { entity: entityFromUrl } = convertURLToMetricsExplore(
+          url.searchParams,
+          AD_BIDS_METRICS_3_MEASURES_DIMENSIONS,
+          explore,
+          basePreset,
+        );
+
+        // assert that the entity we got back matches the expected entity
+        expect(entityFromUrl).toEqual({
+          ...initEntity,
+          ...entity,
+        });
+
+        // go back to default url
+        const defaultUrl = new URL("http://localhost");
+        const { entity: entityFromDefaultUrl } = convertURLToMetricsExplore(
+          defaultUrl.searchParams,
+          AD_BIDS_METRICS_3_MEASURES_DIMENSIONS,
+          explore,
+          basePreset,
+        );
+
+        // assert that the entity we got back matches the original
+        expect(entityFromDefaultUrl).toEqual(initEntity);
+      });
+    }
+  });
+
+  describe("Should set correct state for legacy protobuf state and restore default state on empty params", () => {
+    for (const { title, entity, preset } of TestCases) {
+      it(title, () => {
+        const url = new URL("http://localhost");
+        const explore: V1ExploreSpec = {
+          ...AD_BIDS_EXPLORE_INIT,
+          ...(preset ? { defaultPreset: preset } : {}),
+        };
+        const basePreset = getBasePreset(explore, {
+          timeZone: "UTC",
+        });
+        const initEntity = getDefaultMetricsExplorerEntity(
+          AD_BIDS_EXPLORE_NAME,
+          AD_BIDS_METRICS_3_MEASURES_DIMENSIONS,
+          explore,
+          AD_BIDS_TIME_RANGE_SUMMARY,
+        );
+        cleanMetricsExplore(initEntity);
+        // load url with legacy protobuf state
+        url.searchParams.set(
+          "state",
+          getProtoFromDashboardState({
+            ...initEntity,
+            ...entity,
+          }),
+        );
+
+        // get back the entity from url params
+        const { entity: entityFromUrl } = convertURLToMetricsExplore(
+          url.searchParams,
+          AD_BIDS_METRICS_3_MEASURES_DIMENSIONS,
+          explore,
+          basePreset,
+        );
+        // assert that the entity we got back matches the expected entity
+        expect(entityFromUrl).toEqual({
+          ...initEntity,
+          ...entity,
+        });
+
+        // go back to default url
+        const defaultUrl = new URL("http://localhost");
+        const { entity: entityFromDefaultUrl } = convertURLToMetricsExplore(
+          defaultUrl.searchParams,
+          AD_BIDS_METRICS_3_MEASURES_DIMENSIONS,
+          explore,
+          basePreset,
+        );
+
+        // assert that the entity we got back matches the original
+        expect(entityFromDefaultUrl).toEqual(initEntity);
+      });
+    }
+  });
+});
+
+// cleans up any UI only state from MetricsExplorerEntity
+export function cleanMetricsExplore(
```

Comment:
This comment implies the function name could be `removeUIStateFromMetricsExplorerEntity`. But doesn't the `MetricsExplorerEntity` contain _all_ UI/client-side state?